pymatgen/entries/compatibility.py

```diff
@@ -242,84 +287,291 @@ def get_correction(self, entry) -> float:
+                        )
+                    elif ox_type == "hydroxide":
+                        correction += (
+                            ufloat(
+                                self.comp_correction["oxide"], self.comp_errors["oxide"]
+                            )
+                            * comp["O"]
+                        )
+                else:
+                    warnings.warn(
+                        "No structure or oxide_type parameter present. Note "
+                        "that peroxide/superoxide corrections are not as "
+                        "reliable and relies only on detection of special"
+                        "formulas, e.g., Li2O2."
+                    )
+                    rform = entry.composition.reduced_formula
+                    if rform in UCorrection.common_peroxides:
+                        correction += (
+                            ufloat(
+                                self.comp_correction["peroxide"],
+                                self.comp_errors["peroxide"],
+                            )
+                            * comp["O"]
+                        )
+                    elif rform in UCorrection.common_superoxides:
+                        correction += (
+                            ufloat(
+                                self.comp_correction["superoxide"],
+                                self.comp_errors["superoxide"],
+                            )
+                            * comp["O"]
+                        )
+                    elif rform in UCorrection.ozonides:
+                        correction += (
+                            ufloat(
+                                self.comp_correction["ozonide"],
+                                self.comp_errors["ozonide"],
+                            )
+                            * comp["O"]
+                        )
+                    elif Element("O") in comp.elements and len(comp.elements) > 1:
+                        correction += (
+                            ufloat(
+                                self.comp_correction["oxide"], self.comp_errors["oxide"]
+                            )
+                            * comp["O"]
+                        )
+            else:
+                correction += (
+                    ufloat(self.comp_correction["oxide"], self.comp_errors["oxide"])
+                    * comp["O"]
+                )
+
+        for anion in ["Br", "I", "Se", "Si", "Sb", "Te", "H", "N", "F", "Cl"]:
+            if Element(anion) in comp and anion in self.comp_correction:
+                apply_correction = False
+                # only apply anion corrections if the element is an anion
+                # first check for a pre-populated oxidation states key
+                # the key is expected to comprise a dict corresponding to the first element output by
+                # Composition.oxi_state_guesses(), e.g. {'Al': 3.0, 'S': 2.0, 'O': -2.0} for 'Al2SO4'
```

Comment:
For consistency with how the old `MPCompatibility` works, we've built this in a way that's strictly composition-based. I recall having some conversations about a more structure-based scheme, but we decided that it would be too complicated / not worth the effort at this point. We could add an extra layer of logic to look for oxidation states when we have a  `ComputedStructureEntry` if you think that would be beneficial. That's also an enhancement we can add later as we integrate more of the oxi state information into the builders.